packages/mui-styled-engine-sc/src/index.d.ts

```diff
@@ -272,6 +273,26 @@ export interface ThemedStyledFunctionBase<
   ): StyledComponent<C, T, O & U, A>;
 }
 
+type RuleSet<Props> = Interpolation<Props>[];
+
+declare function css(
+  styles:
+    | TemplateStringsArray
+    | CSSObject
+    | InterpolationFunction<ThemedStyledProps<object, DefaultTheme>>,
```

Comment:
Won't using `DefaultTheme` in this type mean that the theme in the css prop will have the wrong type? It won't be the Material UI's theme. Ideally I feel like we should export from here a "generator" for this type and re-exprot it from Material UI using the Material UI's Theme. We would do the same for the other package as well (@mui/styled-engine).

Reply:
I basically agree. That is what my [other PR](https://github.com/mui/material-ui/pull/40656/files) is actually doing/solving.

The issue here is, the actual theme variable we receive within the css-function, is not the material theme. We have to use `theme[THEME_ID]` in order to access the MUI theme. So this problem already exists currently. But at least, with this fix/PR, we have solved the typing issue for variables. Step 2 would be getting the value and types of the theme prop right (which is solved in the other PR mentioned above).

Reply:
As long as we do not fix the theme variable thing properly (other PR), i do not know what other type we could use here.

Reply:
@delijah I got your point. Would it make sense to do it in the same PR? This way we prevent the scenario where 2 PRs are not in the same release.

From what I see, both of your PRs do not contain huge changes so I think they can be combined.